src/Microsoft.OpenApi.OData.Reader/Common/EdmModelHelper.cs

```diff
@@ -341,5 +343,51 @@ internal static string GenerateODataTypeCastPathOperationIdPrefix(ODataPath path
 
             return operationId;
         }
+
+        /// <summary>
+        /// Strips or aliases namespace prefixes from an element name.
+        /// </summary>
+        /// <param name="element">The target element.</param>
+        /// <param name="model">Optional: The Edm model. Used for searching for the namespace alias.</param>
+        /// <param name="settings">The OpenAPI convert settings.</param>
+        /// <returns>The element name, alias-prefixed or namespace-stripped if applicable.</returns>
+        internal static string StripOrAliasNamespacePrefix(IEdmSchemaElement element, OpenApiConvertSettings settings, IEdmModel model = null)
+        {
+            Utils.CheckArgumentNull(element, nameof(element));
+            Utils.CheckArgumentNull(settings, nameof(settings));
+
+            string namespaceAlias = string.Empty;
+            string namespaceName = element.Namespace;
+            string segmentName = element.FullName();        
+
+            if (!string.IsNullOrEmpty(namespaceName) && model != null)
+            {
+                namespaceAlias = model.GetNamespaceAlias(namespaceName);
+            }         
+
+            if (element is IEdmStructuredType && settings.EnableAliasForTypeCastSegments && !string.IsNullOrEmpty(namespaceAlias))
+            {
+                // Alias type cast segment name
+                segmentName = namespaceAlias.TrimEnd('.') + "." + element.Name;
+            }
+            
+            if (element is IEdmOperation)
+            {                
+                if (settings.EnableAliasForOperationSegments && !string.IsNullOrEmpty(namespaceAlias))
+                {
+                    // Alias operation segment name 
+                    segmentName = namespaceAlias.TrimEnd('.') + "." + element.Name;
+                }
+                
+                if (settings.NamespacePrefixToStripForInMethodPaths != null && 
```

Comment:
```suggestion
                else if (!string.IsNullOrEmpty(settings.NamespacePrefixToStripForInMethodPaths) && 
```

Didn't we decide not to strip prefix when we have an alias? without the else both conditions could apply even though their body is definitive as of today

Reply:
@baywet I believe both conditions would only apply when the method is in the root namespace and the root namespace has an alias as well. 
In which case, we would rather have no prefix. Changing to the `else if` would lead to `microsoft.graph.add` to become `graph.add`(root namespace has the alias `graph`) rather than `add`. Yes?

Reply:
+1
 
cc @irvinesunday

Reply:
Coolio